### PR TITLE
Add manual testing docs for Indirect Diffraction

### LINKED ISSUE #37416: Add manual testing instructions for Indirect Diffraction
**Describe the outcome that is desired.**
We should add manual testing instructions on for the Indirect Diffraction interface so we can catch more issues during manual testing.

The new page should be accessible from here:
https://developer.mantidproject.org/Testing/index.html

**Acceptance Criteria**
- [ ] Indirect Diffraction has its own manual testing instructions

### PR DESCRIPTION
**Pull request by robertapplin**
### Description of work
This PR adds manual testing instructions for the Indirect Diffraction interface.

Fixes #37416 

### To test:
Follow the instructions in the new documentation page. Make sure they are clear and that they make sense

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

**Comment by jhaigh0** [comment]
Might be doing something wrong, but when I click Run for the first time on the first test I'm getting an unhandled exception
```
AlgorithmManager-[Error] AlgorithmManager:: Unable to create algorithm ISISDiffractionReduction AlgorithmFactory::highestVersion() - Unknown algorithm 'ISISDiffractionReduction'
```
and then a crash. I'm testing on the current main build.

**Comment by robertapplin** [comment]
> Might be doing something wrong, but when I click Run for the first time on the first test I'm getting an unhandled exception
> 
> ```
> AlgorithmManager-[Error] AlgorithmManager:: Unable to create algorithm ISISDiffractionReduction AlgorithmFactory::highestVersion() - Unknown algorithm 'ISISDiffractionReduction'
> ```
> 
> and then a crash. I'm testing on the current main build.

Ah yes, I needed to rebase the branch on main so I get the changes from https://github.com/mantidproject/mantid/pull/37472. It should be working now

File: dev-docs/source/Testing/Indirect/DiffractionTests.rst [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+#. Open ``Interfaces`` > ``Indirect`` > ``Diffraction``
+#. Make sure ``Instrument`` is set to ``IRIS``
+#. The default Detector Grouping should be ``All``
+#. In ``Run Numbers`` enter 26176
+#. Click ``Run``
+#. The output workspace should have 1 spectra
+#. Change the Detector Grouping too ``Groups`` and enter 4 groups
+#. Click ``Run``
+#. The output workspace should have 4 spectra
+#. Change the Detector Grouping too ``Custom``, enter ``105-108,109-112``
```
Comment (jhaigh0) [inline dev-docs/source/Testing/Indirect/DiffractionTests.rst:28]:
```suggestion
#. Change the Detector Grouping to ``Groups`` and enter 4 groups
#. Click ``Run``
#. The output workspace should have 4 spectra
#. Change the Detector Grouping to ``Custom``, enter ``105-108,109-112``
```

File: dev-docs/source/Testing/Indirect/DiffractionTests.rst [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+#. Using ``Groups``, try more than 8 groups. This should not be allowed.
+#. Using ``File``, try an empty string. This should not be allowed.
+
+**Time required 2-5 minutes**
+
+--------------
+
+#. Open ``Interfaces`` > ``Indirect`` > ``Diffraction``
+#. Make sure ``Instrument`` is set to ``OSIRIS``
+#. Set ``Analyser`` to ``Diffonly``
```
Comment (jhaigh0) [inline dev-docs/source/Testing/Indirect/DiffractionTests.rst:41]: the Reflection combo box has this option.
```suggestion
#. Set ``Reflection`` to ``Diffonly``
```

File: dev-docs/source/Testing/Indirect/DiffractionTests.rst [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+#. In ``Cal File`` load the ``osiris_041_RES10.cal`` file from the unit test data
+#. Select Detector Grouping to be ``All``
+#. Click ``Run``
+#. The output workspace should have 1 spectra
+#. Change the Detector Grouping too ``Groups`` and enter 5 groups
+#. Click ``Run``
+#. The output workspace should have 5 spectra
+#. Change the Detector Grouping too ``Custom``, enter ``3-500,501-962``
+#. Click ``Run``
+#. The output workspace should have 2 spectra
```
Comment (jhaigh0) [inline dev-docs/source/Testing/Indirect/DiffractionTests.rst:53]: This makes reference to one output workspace, but I get three ending in `_dRange`, `_q`, `_tof`. The second two changed to five spectra, then they all changed to two.

**Review by jhaigh0 — CHANGES_REQUESTED** [review]
A few small things but pretty much all there.

File: dev-docs/source/Testing/Indirect/DiffractionTests.rst (hunk 3 above)
Comment (robertapplin) [reply dev-docs/source/Testing/Indirect/DiffractionTests.rst]: Thanks, I've made those changes by ammending the previous commit.

When I test locally, all three of the output workspaces have the same number of spectra for each step (e.g. 1, then 5, then 2). Are you seeing `_dRange` doesn't have 5 spectra when using ``Groups`` option?

Comment (jhaigh0) [reply dev-docs/source/Testing/Indirect/DiffractionTests.rst]: I thought not but I was working quickly so very possible it just hadn't updated yet or I misread.

File: dev-docs/source/Testing/Indirect/DiffractionTests.rst [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+#. In ``Vanadium File`` enter 89757
+#. In ``Cal File`` load the ``osiris_041_RES10.cal`` file from the unit test data
+#. Select Detector Grouping to be ``All``
+#. Click ``Run``
+#. There should be three output workspaces, ending in ``_dRange``, ``_q`` and ``_tof``
+#. Each of the output workspaces should have 1 spectra
+#. Change the Detector Grouping too ``Groups`` and enter 5 groups
+#. Click ``Run``
+#. Each of the output workspaces should have 5 spectra
+#. Change the Detector Grouping too ``Custom``, enter ``3-500,501-962``
```
Comment (jhaigh0) [inline dev-docs/source/Testing/Indirect/DiffractionTests.rst:52]:
```suggestion
#. Change the Detector Grouping to ``Groups`` and enter 5 groups
#. Click ``Run``
#. Each of the output workspaces should have 5 spectra
#. Change the Detector Grouping to ``Custom``, enter ``3-500,501-962``
```

File: dev-docs/source/Testing/Indirect/DiffractionTests.rst (hunk 3 above)
Comment (robertapplin) [reply dev-docs/source/Testing/Indirect/DiffractionTests.rst]: Ah maybe, I can't seem to replicate it 🤷‍♂️

**Review by jhaigh0 — APPROVED** [review]

**Pull request merged**